src/core/main.c

```diff
@@ -147,9 +149,52 @@ static NUMAPolicy arg_numa_policy;
 /* A copy of the original environment block */
 static char **saved_env = NULL;
 
+static volatile sig_atomic_t flag_self_healing_reexec = 0;
```

Comment:
We don't say "flag_" in general. Maybe call this "waiting_for_signal_to_reexec" or similar.